File: examples/extending-objects/Inheritance/Kotlin/Person.kt
// Assuming everything is in the same package

open class Person (firstName : String, lastName : String, age : Int) {
  val firstName : String
  val lastName : String
  var age : Int

  init {
	this.firstName = firstName
	this.lastName = lastName
	this.age = age   
  }
  
  open fun toString() : String {
    return this.firstName + " " + this.lastName + ", " + this.age;
  }
}

File: examples/extending-objects/Inheritance/C++/Person.cpp
#include <string>
using namespace std;

class Person {
  protected:
  	string firstName;
  	string lastName;
  	int age;
    
  public:
  	Person(string firstName, string lastName, int age) {
  		this->firstName = firstName;
  		this->lastName = lastName;
  		this->age = age;
  	}
	
    string toString() {
		return this->firstName + " " + this->lastName + ", " + to_string(this->age);
	}
};